### DISCUSSION TITLE
Create console that can be resized

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#11430] [General - Components / Directives / etc] [answered]
How can I make it so when the console is visible it fills the available vertical space, but when enabled, the console fills the % of amount defined by the user clicking and dragging the resize bar? This feature itself exists within CodePen which is cool, im trying to do something similar for my app but just with a main body section and console. I've got a codepen showing what im trying to do. 

https://codepen.io/jokermartini/pen/yLorBeJ?editors=1011

Right now i get this

![image](https://user-images.githubusercontent.com/1416749/142737397-7458e583-9b79-481f-874e-c38d26834f21.png)

It want it to full the full height  like this

![image](https://user-images.githubusercontent.com/1416749/142737410-ba83bd60-5015-45b0-a2d5-a0d21bf42abb.png)

Then when the console is hidden the body should fill the full height. Keep in mind the main body and console show their own scrollbars if needed.

![image](https://user-images.githubusercontent.com/1416749/142737429-7b3f9484-0009-4448-9dc0-9f356349feb3.png)

**Comment by suleimanas** [comment]
you are over wrapping your divs, try to make it simple :)
```
<q-page class="column">
  <div id="section_1" class="col bg-red">full-height</div>
  <div id="console" class="col-auto bg-blue">auto-height</div>
</q-page>
```

**Reply by JokerMartini** [reply]
thanks, i felt like i was over complicating it :)

**Reply by JokerMartini** [reply]
After doing that though, i still have the same problems i originally had, how do i control the overflow so it doesn't push the contents down. i want the areas to always fit within the view.

```
<q-page class="column">
    <div id="section_1" class="col bg-red scroll">
      full-height
      <div class="bg-purple" style="overflow: auto;">
        <div class="text-h4 q-mb-md">Before</div>
        <div v-for="n in 50" :key="n" class="q-my-md">
          {{ n }}. Lorem ipsum dolor sit, amet consectetur adipisicing elit.
          Quis praesentium cumque magnam odio iure quidem, quod illum numquam
          possimus obcaecati commodi minima assumenda consectetur culpa fuga
          nulla ullam. In, libero.
        </div>
      </div>
    </div>
    <div
      id="console"
      v-if="isCategoriesEnabled"
      class="col-auto bg-blue"
      style="height: 200px"
    >
      <div
        style="height: 6px; cursor: n-resize"
        class="bg-orange"
        v-touch-pan.vertical.prevent.mouse="handlePan"
      ></div>
      <div style="background: red; overflow: auto;">
        <div v-for="n in 6" :key="n" class="q-my-md">
          {{ n }}. Lorem ipsum dolor sit, amet consectetur adipisicing elit.
          Quis praesentium cumque magnam odio iure quidem, quod illum numquam
          possimus obcaecati commodi minima assumenda consectetur culpa fuga
          nulla ullam. In, libero.
        </div>
      </div>
    </div>
  </q-page>
```

**Comment by suleimanas** [comment, marked as answer]
classes to use: **relative-position** on the col section and then **absolute-full scroll overflow-auto**

[https://codepen.io/suleimanas/pen/mdMgmXg?editors=1010]

```
<q-page class="column">
  <div class="col bg-purple relative-position">
    <div class="absolute-full scroll overflow-auto">
        <div class="text-h4 q-mb-md">Before</div>
        <div v-for="n in 50" :key="n" class="q-my-md">
          {{ n }}. Lorem ipsum dolor sit, amet consectetur adipisicing elit.
          Quis praesentium cumque magnam odio iure quidem, quod illum numquam
          possimus obcaecati commodi minima assumenda consectetur culpa fuga
          nulla ullam. In, libero.
        </div>
    </div>
  </div>

  <div
      v-if="isCategoriesEnabled"
      class="col-auto bg-blue relative-position"
      style="height: 200px"
  >
    <div class="absolute-full scroll overflow-auto">
         <div v-for="n in 6" :key="n" class="q-my-md">
          {{ n }}. Lorem ipsum dolor sit, amet consectetur adipisicing elit.
          Quis praesentium cumque magnam odio iure quidem, quod illum numquam
          possimus obcaecati commodi minima assumenda consectetur culpa fuga
          nulla ullam. In, libero.
        </div>
    </div>
  </div>
</q-page>
```

**Reply by JokerMartini** [reply]
You are a genius!!! thank you so much

**Reply by suleimanas** [reply]
glad to help, and shoutout to @rstoenescu he's the genius who made it that easy :)

**Reply by JokerMartini** [reply]
Much agreed. I'll try to post my final code once i get it working. I'm having some slight issues with an aspect of it. but it's so close. ill make a fiddle

**Reply by JokerMartini** [reply]
@suleimanas  we are so soooo close. I'm just having issues with the Console header/bar sticking to the top of that section, without scrolling. You'll see in the code that i have two options which the **Option 1** for some reason does not acknowledge the mouse event, however **Option 2** mouse event works but scrolls with the console content which is not desired. 

https://codepen.io/jokermartini/pen/yLorBeJ

**Reply by JokerMartini** [reply]
I got it i just had to move the Grip outside of the col all together. That did the trick

**Reply by suleimanas** [reply]
Awesome, and why don't you check the splitter component also
https://quasar.dev/vue-components/splitter

**Reply by JokerMartini** [reply]
I tried using the splitter component but had some issues being able to completely disable a section of it. I couldn't do a conditional check within the template to remove the 'Console' section. Otherwise I would have used it.

**Reply by JokerMartini** [reply]
If you can find a way to do that im all for it, but i couldn't figure out a nice way to get it to work.
